### PR TITLE
LPS-186422 allow filtering search results

### PR DESCRIPTION
**Pull request by marcogalluzzi**
See issue for detailed requirements: https://issues.liferay.com/browse/LPS-129663

There are many commits but they are easy to follow.

The "view.jsp" and the "search_resources.jsp" in Document Library are quite similar. I would unify both files in another PR.

### CODE REVIEW

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Comment by marcogalluzzi** [comment]
Hi @ambrinchaudhary can you please make the changes required by the source formatter? I suspect that they are not as simple as replacing one tag with another and it is probably easier for you. If you don't have time I can do it myself. Thanks.

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java [hunk 1]
```diff
@@ -1228,6 +1113,73 @@ private boolean _isStagedActions() {
...
+		long[] assetCategoryIds = ArrayUtil.toLongArray(
+			_getSelectedAssetCategoryIds());
+
+		if (ArrayUtil.isNotEmpty(assetCategoryIds)) {
+			portletURL.setParameter(
+				"assetCategoryId", ArrayUtil.toStringArray(assetCategoryIds));
+		}
+		else {
+			portletURL.setParameter("assetCategoryId", (String)null);
+		}
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:1126]: Why do you need to check for emptiness here? If the array is empty, ArrayUtil.toStringArray will return the empty string (not null)? Do you need this to be null for some specific reason? The same applies to all other usages here.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java [hunk 2]
```diff
@@ -431,10 +430,8 @@ public boolean isRootFolderNotFound() {
 	}
 
 	public boolean isSearch() {
-		String mvcRenderCommandName = ParamUtil.getString(
-			_httpServletRequest, "mvcRenderCommandName");
-
-		return mvcRenderCommandName.equals("/document_library/search");
+		return !Validator.isBlank(
+			ParamUtil.getString(_httpServletRequest, "keywords"));
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java:434]: That's right. I missed it!

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java [hunk 3]
```diff
@@ -934,20 +934,25 @@ private List<RepositoryEntry> _getRepositoryEntries(Hits hits) {
 	}
 
 	private SearchContext _getSearchContext(
-		SearchContainer<RepositoryEntry> searchContainer) {
+		SearchContainer<RepositoryEntry> searchContainer, boolean addSorting,
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java:937]: There are two calls to `_getSearchContext`, I will re-check if there's a an easy and safe way to know who is calling the method to apply the FF condition.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 1 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java:1126]: I'm not sure if `null` value is really needed. I just replicated how it was working before. I need to test if `null` is needed or not.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java]: Done in new commit https://github.com/liferay-lima/liferay-portal/pull/4378/commits/61f9b529e3854f535b115fd42eb3f102e51fd7a8

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java (hunk 1 above)
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminManagementToolbarDisplayContext.java]: After testing I simplified the method by removing all emptiness checks. Done in new commit https://github.com/liferay-lima/liferay-portal/pull/4378/commits/61f9b529e3854f535b115fd42eb3f102e51fd7a8

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java]: Done, I've removed the need to pass the FF as an input parameter. See last commit [fe27011](https://github.com/liferay-lima/liferay-portal/pull/4378/commits/fe27011f8af33775c8603c2445a0fbf365c84942)

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
wh:test

**Comment by adolfopa** [comment]
wh:test

**Comment by adolfopa** [comment]
wh:test

**Comment by adolfopa** [comment]
wh:test

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
wh:test

**Comment by adolfopa** [comment]
wh:test

**Comment by adolfopa** [comment]
wh:test
